### PR TITLE
IDEMPIERE-5567 Support of UUID as Key (FHCA-4195) - Toolbar

### PR DESCRIPTION
**Pull request by CarlosRuiz-globalqss**
https://idempiere.atlassian.net/browse/IDEMPIERE-5567

- add support for AD_Attachment, CM_Chat, AD_PostIt and AD_LabelAssignment for UUID based tables

# Pull Request Checklist

- [x] My code follows the [code guidelines](https://wiki.idempiere.org/en/Contributing_to_Trunk) of this project
- [x] My code follows the best practices of this project
- [x] I have performed a self-review of my own code
- [x] My code is easy to understand and review. 
- [x] I have checked my code and corrected any misspellings
- [x] In hard-to-understand areas, I have commented my code.
- [x] My changes generate no new warnings
### Tests
- [x] I have tested the direct scenario that my code is solving
- [x] I checked all collaterals that can be affected by my changes, and tested other potential affected scenarios
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have added unit tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: org.adempiere.base/src/org/compiere/model/GridTable.java [hunk 1]
```diff
@@ -1043,11 +1043,11 @@ public int getKeyID (int row)
...
 
 	/**
-	 *	Get UUID or null of none
+	 *	Get Key UUID or null of none
 	 *  @param row row
-	 *  @return UUID or null
+	 *  @return ID or null
 	 */
-	public UUID getUUID (int row)
+	public String getKeyUUID (int row)
```
Comment (hengsin) [inline org.adempiere.base/src/org/compiere/model/GridTable.java:1050]: Changing a public API like this break backward compatibility

**Comment by hengsin** [comment]
Looks fine but I think we need to add to the migration note the potential performance (migration wise) and extra storage issue for installation with large amount of attachment records.

**Comment by CarlosRuiz-globalqss** [comment]
> Looks fine but I think we need to add to the migration note the potential performance (migration wise) and extra storage issue for installation with large amount of attachment records.

Thanks @hengsin - ready, technical note added, and new options documented

**Review by hengsin — APPROVED** [review]
tested

**Pull request merged**